lib/snowbridge.ts

```diff
@@ -373,3 +386,37 @@ export function getErrorMessage(err: any) {
   console.error(message, err);
   return message;
 }
+
+export async function assetMetadata(
+  context: Context,
+  env: SnowbridgeEnvironment,
+): Promise<BridgedAssetsMetadata> {
+  const tokens = [
+    ...new Set(
+      env.locations
+        .flatMap((l) => l.erc20tokensReceivable)
+        .map((l) => l.address.toLowerCase()),
+    ),
+  ];
+  const [relaychainNativeAsset, assetMetadataList] = await Promise.all([
+    assets.parachainNativeAsset(context.polkadot.api.relaychain),
+    Promise.all(
+      tokens.map((t) =>
+        assets
+          .assetErc20Metadata(context, t)
+          .then((m) => ({ token: t, metadata: m }))
+          .catch((_) => null),
+      ),
+    ),
+  ]);
+
+  const erc20Metadata: { [tokenAddress: string]: assets.ERC20Metadata } = {};
+  assetMetadataList
```

Comment:
Why not filter first?

Reply:
I made this code imperative to show what it does better. It is trying to make a hashmap of token address to metadata given a list of metadata. The metadata may be null in the case of failure, in which case it should report error and skip. As you pointed out I forgot to log error above so nice catch there. Why the filter happens right at the end is because we do the token fetching in parralell.

addressed in [0cade12](https://github.com/Snowfork/snowbridge-app/pull/26/commits/0cade12ec8e6e4ee11f96ac5726be24e5759d327)